src/main/scala/org/scalacheck/MasterRunner.scala

```diff
@@ -0,0 +1,67 @@
+package org.scalacheck
+
+
+import sbt.testing._
+import java.util.concurrent.atomic.AtomicInteger
+
+class  MasterRunner(
+                    args: Array[String],
+                    remoteArgs: Array[String],
+                    loader: ClassLoader)
+  extends BaseRunner(args, remoteArgs, loader) {
+
+
+  /** Number of tasks registered in the whole system */
+  private[this] val registeredCount = new AtomicInteger(0)
+
+  /** Number of running slaves in the whole system */
+  private[this] val slaveCount = new AtomicInteger(0)
+
+  def tasks(taskDefs: Array[TaskDef]): Array[Task] = {
+    registeredCount.addAndGet(taskDefs.length)
+    taskDefs.map(newTask)
+  }
+
+  def done(): String = {
+    val slaves = slaveCount.get
+    val registered = registeredCount.get
+    val done = results.doneCount.get
+    val test = results.testCount.get
+    val success = results.successCount.get
+    val failed = results.failureCount.get
+    val errors = results.errorCount.get
+
+    if (slaves > 0) {
+      throw new IllegalStateException(
+        s"There are still $slaves slaves running")
+    }
+
+    if (registered != done)
+      throw new IllegalStateException(
+        s"$registered task(s) were registered, $done were executed")
+    else {
+      val heading = if (test == success) "Passed" else "Failed"
+      s"$heading: Total $test, Failed $failed, Errors $errors, Passed $success"
+    }
+  }
+
+  private[scalacheck] def taskDone(): Unit = results.addTask()
+
+  private[scalacheck] def testDone(status: Status): Unit =  results.addTest(status)
+
+  def receiveMessage(msg: String): Option[String] = msg(0) match {
+    case 's' =>
+      slaveCount.incrementAndGet()
+      // Send Hello message back
+      Some("Hello")
+    case 't' =>
+      // Slave notifies us of registration of tasks
+      registeredCount.addAndGet(msg.tail.toInt)
+      None
+    case 'd' =>
+      // Slave notifies us of completion of tasks
+      results.add(Results(msg.tail))
```

Comment:
Ah, I think that because you want to do this, you need a master and a slave. But besides that message to rather the `Results`, it seems you can get rid of all the other messages.